File: scripts/snapshot-git.py
#!/usr/bin/env python3
import sys
import json
import os
import subprocess
from pathlib import Path

sys.path.insert(0, os.path.dirname(__file__))
from core.filesystem import get_snapshot
from adapter.bridge.context import hook_entrypoint

@hook_entrypoint(fallback_result={"injectSteps": []})
def main(context):
    transcript_path = context.get('transcriptPath', '')
    cwd = context.get('cwd', os.getcwd())
    
    if not transcript_path:
        return {"injectSteps": []}
        
    try:
        conv_dir = Path(transcript_path).parent.parent.parent
        scratch_dir = conv_dir / 'scratch'
        scratch_dir.mkdir(parents=True, exist_ok=True)
        snapshot_file = scratch_dir / 'remora_pre_snapshot.json'
        
        snapshot = get_snapshot(cwd)
        with open(snapshot_file, 'w', encoding='utf-8') as f:
            json.dump(snapshot, f)
            
    except Exception:
        pass
        
    return {"injectSteps": []}

if __name__ == "__main__":
    main()

File: packages/adapter-antigravity/src/hooks/snapshot-git.ts
import { PreInvocationResponse } from "../types";
import * as fs from "node:fs";
import * as path from "node:path";
import { getSnapshot } from "../bridge/filesystem";

export function main(context: Record<string, any>): PreInvocationResponse {
	try {
		return _main(context);
	} catch {
		return { injectSteps: [] };
	}
}

function _main(context: Record<string, any>): { injectSteps: never[] } {
	const transcriptPath = context["transcriptPath"] ?? "";
	const cwd = context["cwd"] ?? process.cwd();

	if (!transcriptPath) {
		return { injectSteps: [] };
	}

	try {
		const convDir = path.dirname(path.dirname(path.dirname(transcriptPath)));
		const scratchDir = path.join(convDir, "scratch");
		fs.mkdirSync(scratchDir, { recursive: true });
		const snapshotFile = path.join(scratchDir, "remora_pre_snapshot.json");

		const snapshot = getSnapshot(cwd);
		fs.writeFileSync(snapshotFile, JSON.stringify(snapshot), "utf-8");
	} catch {
		// pass
	}

	return { injectSteps: [] };
}

import { hookEntrypoint } from "../bridge/context";

if (typeof require !== "undefined" && require.main === module) {
	hookEntrypoint()(main)();
}
